src/utils/generation_utils.py

```diff
@@ -452,3 +452,54 @@ def await_streamed_block(block: Block, context: AgentContext) -> Block:
         block = Block.get(block.client, _id=block.id)
     context.chat_history.file.refresh()
     return block
+
+
+def generate_action_choices(context: AgentContext) -> Block:
+
+    game_state = get_game_state(context)
+    quest_name = game_state.current_quest
+
+    prompt = (
+        f"Generate a multiple choice set of four options for the user to select {game_state.player.name}'s next "
+        f'action. The actions should be relevant to the story and include a single "custom" option that '
+        f"allows the user to provide their own action in the story. "
+        "The generated actions should match the tone and narrative voice of the existing story.\n"
+        "Action choices should be returned as a JSON map that provides a single letter key for each action "
+        "choice, as follows: \n"
+        '{"A": "pet the dog", "B": "launch missiles", "C": "dance the macarana", "D": "custom" }'
```

Comment:
quibbliest of quibbles: Macarena?

Reply:
😆 i'm just happy i got it somewhat close as I was mashing up an example.